feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -392,33 +392,32 @@ class BillingWrapper(
         withConnectedClient {
             log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE)
 
-            val queryActiveSubscriptionsResult = this.queryPurchases(SkuType.SUBS)
-            if (!queryActiveSubscriptionsResult.isSuccessful()) {
-                val billingResult = queryActiveSubscriptionsResult.billingResult
-                val purchasesError = billingResult.responseCode.billingResponseToPurchasesError(
-                    RestoreStrings.QUERYING_SUBS_ERROR.format(billingResult.toHumanReadableDescription())
-                )
-                onError(purchasesError)
-                return@withConnectedClient
-            }
+            this.queryPurchasesAsync(SkuType.SUBS) { activeSubsResult, activeSubsPurchases ->
+                if (!activeSubsResult.isSuccessful()) {
+                    val purchasesError = activeSubsResult.responseCode.billingResponseToPurchasesError(
+                        RestoreStrings.QUERYING_SUBS_ERROR.format(activeSubsResult.toHumanReadableDescription())
+                    )
+                    onError(purchasesError)
+                    return@queryPurchasesAsync
+                } else {
+                    val mapOfActiveSubscriptions = activeSubsPurchases.toMapOfGooglePurchaseWrapper(SkuType.SUBS)
 
-            val queryUnconsumedInAppsResult = this.queryPurchases(SkuType.INAPP)
-            if (!queryUnconsumedInAppsResult.isSuccessful()) {
-                val billingResult = queryUnconsumedInAppsResult.billingResult
-                val purchasesError = billingResult.responseCode.billingResponseToPurchasesError(
-                    RestoreStrings.QUERYING_INAPP_ERROR.format(billingResult.toHumanReadableDescription())
-                )
-                onError(purchasesError)
-                return@withConnectedClient
+                    this.queryPurchasesAsync(SkuType.INAPP) { unconsumedInAppsResult, unconsumedInAppsPurchases ->
+                        if (!unconsumedInAppsResult.isSuccessful()) {
+                            val purchasesError = unconsumedInAppsResult.responseCode.billingResponseToPurchasesError(
+                                RestoreStrings.QUERYING_INAPP_ERROR.format(
+                                    unconsumedInAppsResult.toHumanReadableDescription()
+                                )
+                            )
+                            onError(purchasesError)
+                        } else {
```

Comment:
same here about removing this else to remove indentation and returning after the `onError` in line 412